github/github.go

```diff
@@ -895,12 +904,26 @@ type BasicAuthTransport struct {
 
 // RoundTrip implements the RoundTripper interface.
 func (t *BasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	req = cloneRequest(req) // per RoundTrip contract
-	req.SetBasicAuth(t.Username, t.Password)
+	// To set extra headers, we must make a copy of the Request so
+	// that we don't modify the Request we were given. This is required by the
+	// specification of http.RoundTripper.
+	//
+	// Since we are going to modify only req.Header here, we only need a deep copy
+	// of req.Header.
+	// shallow copy of the struct
+	req2 := new(http.Request)
+	*req2 = *req
+	// deep copy of the Header
```

Comment:
Also get rid of "shallow copy of the struct" and "deep copy of the Header" here.